src/OrchardCore.Modules/OrchardCore.Flows/Views/BagPart.cshtml

```diff
@@ -8,6 +8,9 @@
     {
         var itemContent = await ContentItemDisplayManager.BuildDisplayAsync(item, Model.BuildPartDisplayContext.Updater, Model.Settings.DisplayType ?? "Detail", Model.BuildPartDisplayContext.GroupId);
 
+        itemContent.Metadata.Alternates.Add("Content_Contained");
+        itemContent.Metadata.Alternates.Add($"Content_Contained__{item.ContentItem.ContentType}");
+
```

Comment:
Doing it like this makes the `Content_Contained` alternate takes precedence over all the normal alternates, like `Content_MyBagItem`, which we can't do.

@jtkech may have some more advice on how the alternate precedence works.

Possibly
```
itemContent.Metadata.OnProcessing(shape => {
         shape.Metadata.Alternates.Add("Content_Contained");
        shape.Metadata.Alternates.Add($"Content_Contained__{item.ContentItem.ContentType}");
        return Task.FromResult(shape);
        });
```

Might work, but I would need @jtkech to confirm this will make it last, in terms of precedence

Reply:
So, `Contained` is a kind of **display mode**,

@deanmarcussen so you would want the following reverse precedence (alternates order)

    Content` // <= in fact not an alternate, the actual shape type on which we fallback

    // The alternate we add but with a lower precedence than 'Content__MyBagItem'
    Content_Contained

    // Normal alternate added in 'ContentItemDisplayManager.BuildDisplayAsync()'
    Content__MyBagItem

    // The normal alternates added in 'Shapes.cs' in '("Content").OnDisplaying()'
    Content_Detail
    Content__0 // <= here 0 is the contentItem.Id, equals to 0 as it is in a bag
    Content_Detail__MyBagItem
    Content_Detail__0 <= here 0 is the contentItem.Id, equals to 0 as it is in a bag

    // The other alternate we add with the higher precedence
    Content_Contained__MyBagItem <= the more specialized one

So specifying the content type would have an higher weight than specifying the `Contained` "display mode". Notice that this is not the case for the display type, `Content_Detail` is higher than `Content__MyBagItem`. So maybe normal that, if we already have `Content-MyBagItem.cshtml`, we would also need `Content-MyBagItem.Contained.cshtml`.

But makes sense that `Content_Contained` only overrides `Content` but not `Content__MyBagItem`.

---

**Do we want to use the display type?**  If so do you remember what we had to do with this kind of alternate

    {fieldType}_{ctx.Shape.Metadata.DisplayType}__{displayMode}{DisplayToken}

If so we would need to add e.g

    Content_Contained__ContentType` // <= if no display type or == "Detail"
    Content_DisplayType__ContentType_Contained` // <= use the display type and _Contained at the end

    Content_Contained` // <= if no display type or == "Detail"
    Content_DisplayType_Contained` // <= use the display type and _Contained at the end

Because only the term after the 1st `_` is moved at the end of the file name and by using a `.`

**Do we want to use the stereotype?** Do we want to use the content item id? But here always equals to 0 ;)

Maybe we would need to do what is done in `ContentItemDisplayManager` that checks / uses both **the display type and the stereotype to compute the actual shape type**. And then add the same alternates, as done in `Shape.cs` in `("Content").OnDisplaying()`, but with the `Contained` display mode.

---

Maybe we miss a display mode metadata property, so that the right alternates could be added in the right order by the existing `Content` shape. Maybe **we could check if the contentitem id** is equal to 0, hmm not a good idea.

But to keep it simple, maybe **just only add these 2 alternates and in the right order**, hmm but the display type may be important, and also the stereotype that may implies another shape type if not null, will see tomorrow.

The `.OnProcessing()` event is a good idea because it executes after the `("Content").OnDisplaying()` so we can see all the normal alternates which is not the case before in the bagpart view. But it doesn't solve yet the order issue because we can't insert alternates, just add them at the end if they don't already exist.

So without checking the stereotype and the display type, with what we have i would suggest

    @using OrchardCore.DisplayManagement.Shapes;
    ...

        var alternates = new AlternatesCollection("Content_Contained");
        alternates.AddRange(itemContent.Metadata.Alternates);
        alternates.Add($"Content_Contained__{item.ContentItem.ContentType}");

        itemContent.Metadata.Alternates.Clear();
        itemContent.Metadata.Alternates.AddRange(alternates);

Which results with the following alternates, looks good

    "Content_Contained"
    "Content__MyBagItem"
    "Content_Detail"
    "Content__0"
    "Content_Detail__MyBagItem"
    "Content_Detail__0"
    "Content_Contained__MyBagItem"

Okay just tried by specifying `Summary` in the bag part settings, which results with

    "Content_Contained" // here we would need "Content_Summary_Contained"
    "Content_Summary__BagItem" // <= only the normal alternate use the display type
    "Content_Contained__BagItem" // here we would need "Content_Summary__BagItem_Contained"

And tried by specifying the `RawHtml` widget type for the contained type, which results with


    "Content_Contained" // here we would need "Widget_Contained"
    "Widget__RawHtml" // <= only the normal alternate use the stereotype
    "Content_Contained__RawHtml" // here we would need "Widget_Contained__RawHtml"

So as said above, to be consistent maybe we would have to check the display type and the stereotype

Reply:
Thank you @jtkech you have both cleared my question up, and given me another hint about what the actual issue is.

> **Do we want to use the display type?**

What I realise is that the issue we are trying to fix here is a regression from a previous change to fix another issue. The original issue was https://github.com/OrchardCMS/OrchardCore/issues/4372 and the fix went in quickly just before RC1 https://github.com/OrchardCMS/OrchardCore/pull/4376/files

I think the fix we put in has created two issues, i.e. the wrong template picked up by `TheAgencyTheme` and the `PageTitle` being messed up.

By changing the default display type to `Detail` we are now, by default, treating items that are contained as it they are standalone (Detail) content items. When actually they are not `Detail` items, they are `Summary` items.

By treating them as `Detail` items the `Content.liquid` template in `TheAgencyTheme` is being picked up as the default template.

Because of this the title is being rendered in the `Header` zone as defined by that template.

In addition to that the `ContentsDriver` is rendering the `ContentsMetadata` shape for the `Detail` display type. This causes the issue that @microposmp has fixed by changing the `PageTitleBuilder`

If the default display type for the `BagPart` is changed back to `Summary` then all of the problems go away.

The default `Content.Summary` template renders the content item, the `ContentsMetadata` is not rendered, so does not impact the `PageTitle`, and there is no need for us to try and hack in some alternates at a template level, which when we discussed in triage, was seen as a last resort.

Using the display type of `Summary` (as a default) also makes sense from a `BagPart` routing point of view, as the item displayed in the page, is a `Summary` and the item when routed to is a `Detail` 

Sorry @microposmp I am not trying to sabotage your pr, just trying to figure out what is going on, and what is the best solution :)

Reply:
No problem. I learned a lot about the templating feature when working with this.

I agree to what you suggest about changing back to Summary display type.

Reply:
@deanmarcussen cool!

So we would need a Widget_Summary shape, right?

Or maybe detect that the contained item has a widget stereotype and then display it in the "Detail" display type

Reply:
Yes @jtkech thanks for reminding me to think about display types! 

I think we just need a `Widget_Summary` shape (copied exactly from the existing `Widget` shape), and to change the display type to `Summary` for the `BagPart` as the default.

And then I must argue my case with @sebastienros and @ns8482e !

@microposmp Cool, I'm always learning too. Would you like to do another pr and I am happy to help you with it, or shall I just do one? Perhaps leave this open until the case has been argued ;)

Reply:
@deanmarcussen Thanks, I can do a new PR. Will let you know when it's available.

Reply:
It's not good idea hard-code  alternate inside BagPart